invenio_communities/assets/semantic-ui/js/invenio_communities/community/new.js

```diff
@@ -22,20 +21,56 @@ import {
   Form,
   Message,
 } from "semantic-ui-react";
-import { FieldLabel, RadioField, TextField } from "react-invenio-forms";
+import {
+  FieldLabel,
+  RadioField,
+  TextField,
+  withCancel,
+} from "react-invenio-forms";
 
-import { CommunitiesApiClient } from "./api";
+import { CommunityApi } from "../api";
 import { i18next } from "@translations/invenio_communities/i18next";
+import { communityErrorSerializer } from "../api/serializers";
 
 class CommunityCreateForm extends Component {
   state = {
     error: "",
   };
 
+  componentWillUnmount() {
+    this.cancellableCreate && this.cancellableCreate.cancel();
+  }
+
   setGlobalError = (errorMsg) => {
     this.setState({ error: errorMsg });
   };
 
+  onSubmit = async (values, { setSubmitting, setFieldError }) => {
+    setSubmitting(true);
+    const client = new CommunityApi();
+    const payload = {
+      metadata: {},
+      ...values,
+    };
+    this.cancellableCreate = withCancel(client.create(payload));
+
+    try {
+      const response = await this.cancellableCreate.promise;
+      setSubmitting(false);
+      window.location.href = response.data.links.settings_html;
+    } catch (error) {
+      const { errors, message } = communityErrorSerializer(error);
```

Comment:
please add everywhere:
```
if (error !== 'UNMOUNTED') {
```
if the cmp is unmounted, using `this` will trigger an error. See examples in `invenio-app-ils`.

Reply:
done